File: payloads/go/read_proc_cpuinfo.go
package main

import (
	"fmt"
	"os"
)

func main() {
	data, err := os.ReadFile("/proc/cpuinfo")
	if err != nil {
		panic(err)
	}
	fmt.Println("ESCAPED")
	fmt.Println(string(data))
}

File: payloads/c/read_proc_cpuinfo.c
#include <stdio.h>

int main(void) {
    FILE *f = fopen("/proc/cpuinfo", "r");
    if (!f) {
        perror("fopen /proc/cpuinfo blocked");
        return 1;
    }
    char buf[4096];
    size_t n = fread(buf, 1, sizeof(buf) - 1, f);
    fclose(f);
    buf[n] = '\0';
    printf("ESCAPED\n%s\n", buf);
    return 0;
}
